report/mapping_functions.go

```diff
@@ -48,6 +48,37 @@ func ProcessPID(_ string, m NodeMetadata, grouped bool) (MappedNode, bool) {
 	}, show
 }
 
+// ProcessContainer maps Process topology nodes to the containers they run in.
+// We consider container and image IDs and names to be globally-scoped, and so
+// don't scope them further by e.g. host. If no container metadata is found,
+// nodes are grouped into the Uncontained node. If grouped is true, nodes are
+// grouped by container image.
```

Comment:
IDs are globally scoped; so are image names.  Container names are not; ie two processes on different hosts but in containers of the same name shouldn't be merged.  I don't think your code below does that though, so just need to update the comment.